doc/source/building/index.rst

```diff
@@ -0,0 +1,354 @@
+    ``--no-build-isolation`` flag to the ``pip install`` command, to avoid
+    building against a ``numpy`` wheel from PyPI. In order for that to work you
+    must first install the remaining build dependencies into the conda
+    environment::
+
+      # Either install all SciPy dev dependencies into a fresh conda environment
+      mamba env create -f environment.yml
+
+      # Or, install only the required build dependencies
+      mamba install python numpy cython pythran pybind11 compilers openblas pkg-config
+
+      # To build the latest stable release:
+      pip install scipy --no-build-isolation --no-binary scipy
+
+      # To build a development version, you need a local clone of the SciPy git repository:
+      git clone https://github.com/scipy/scipy.git
+      git submodule update --init
+      pip install . --no-build-isolation
+
+
+.. _the-dev-py-interface:
+
+Building from source for SciPy development
+``````````````````````````````````````````
+
+If you want to build from source in order to work on SciPy itself, first clone
+the SciPy repository::
+
+      git clone https://github.com/scipy/scipy.git
+      git submodule update --init
+
+Then you want to do the following:
+
+1. Create a dedicated development environment (virtual environment or conda
+   environment),
+2. Install all needed dependencies (*build*, and also *test*, *doc* and
+   *optional* dependencies), 
+3. Build SciPy with our ``dev.py`` developer interface.
+
+Step (3) is always the same, steps (1) and (2) are different between conda and
+virtual environments:
+
+.. tab-set::
+
+  .. tab-item:: Virtual env or system Python
+
+    .. note::
+
+       There are many tools to manage virtual environments, like ``venv``,
+       ``virtualenv``/``virtualenvwrapper, ``pyenv``/``pyenv-virtualenv``,
+       Poetry, PDM, Hatch, and more. Here we use the basic ``venv`` tool that
+       is part of the Python stdlib. You can use any other tool; all we need is
+       an activated Python environment.
+
+    Create and activate a virtual environment in a new directory named ``venv``::
+
+       python -m venv venv
+       source venv/bin/activate
+
+    Then install the Python-level dependencies (see ``pyproject.toml``) from
+    PyPI with::
+
+       # Build dependencies
+       python -m pip install numpy cython pythran pybind11 meson ninja pydevtool rich-click
+
+       # Test and optional runtime dependencies
+       python -m pip install pytest pytest-xdist pytest-timeout pooch threadpoolctl asv gmpy2 mpmath
+
+       # Doc build dependencies
+       python -m pip sphinx "pydata-sphinx-theme==0.9.0" sphinx-design matplotlib numpydoc jupytext myst-nb
+
+       # Dev dependencies (static typing and linting)
+       python -m pip mypy typing_extensions types-psutil pycodestyle ruff cython-lint
+
+  .. tab-item:: Conda env
+
+    If you don't have a conda installation yet, we recommend using
+    Mambaforge_; any conda flavor will work though.
```

Comment:
I might have a comment about this. This is a placeholder that I will edit or remove shortly.

Reply:
That must be new, I will check. It always defaulting to asking about adding to the path, and I have `$HOME/mambaforge/bin` as the very first entry on my PATH.

Reply:
It does give the option, but like Anaconda, the option it is uncheck by default and there is a note that recommends against it. (At least on Windows.)
I've always ignored the recommendation in the past and added it, but it might be worth a note.

Reply:
I can't figure it out, there are no docs for Windows it seems, beyond the one-liner at https://github.com/conda-forge/miniforge#windows. @mdhaber would you mind opening an issue on that repo? I don't have a Windows machine and not enough context to open an issue there or respond to questions.

Reply:
Opened conda-forge/miniforge#453.

Reply:
Thanks Matt. First response there seems to indicate it may be a bug. Either way, I think there is nothing to do here - if it stays like that and conda-forge adds docs, we can point to them.